### PR TITLE
fix(tianmu): fix problems related to master slave synchronization (#819)

### PR DESCRIPTION
**Pull request by konghaiya**
fix 1:#818 Master slave synchronization - There will be too many tuples problem

Cause:
Tianmu::dbhandler::TianmuHandler::current_position This variable is not initialized. In some cases, a large value may cause too many tuples problem.
Solution: Initialize the variable

fix 2:#819 Master slave synchronization - Primary key conflict problem

Solution: Modify the modification logic of master slave synchronization, so that the delete and update operations do not follow the primary key logic

Supplement

sql/sql_insert.cc
Fix the problem that the insert statement will not generate binlog in the delayed insert mode storage/tianmu/handler/tianmu_handler.cpp
Fix the binlog error of the line format generated by the tianmu engine

## Summary about this PR

Issue Number: close #issue_number_you_created

## Tests Check List

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

## Changelog

- [ ] New Feature
- [x] Bug Fix
- [ ] Improvement
- [ ] Performance Improvement
- [ ] Build/Testing/CI/CD
- [ ] Documentation
- [ ] Not for changelog (changelog entry is not required)

## Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features

### CODE REVIEW

**Comment by konghaiya** [comment]
CR, PTAL, @RingsC @hustjieke @isredstar @duanfuxiang0

**Review by RingsC — APPROVED** [review]
LGTM

File: storage/tianmu/handler/tianmu_handler.cpp [hunk 1]
```diff
@@ -170,7 +170,23 @@ namespace {
 std::vector<bool> GetAttrsUseIndicator(TABLE *table) {
   int col_id = 0;
   std::vector<bool> attr_uses;
+  enum_sql_command sqlCommand = SQLCOM_END;
+  if (table->in_use && table->in_use->lex)
+    sqlCommand = table->in_use->lex->sql_command;
```
Comment (hustjieke) [inline storage/tianmu/handler/tianmu_handler.cpp]: 1. Uniform var names, refs: 
https://google.github.io/styleguide/cppguide.html#Variable_Names

2. Wrap the expression with brackets

File: storage/tianmu/handler/tianmu_handler.cpp [hunk 2]
```diff
@@ -170,7 +170,23 @@ namespace {
...
+  bool tianmuDeleteOrUpdate = sqlCommand == SQLCOM_DELETE || sqlCommand == SQLCOM_DELETE_MULTI ||
+                              sqlCommand == SQLCOM_UPDATE || sqlCommand == SQLCOM_UPDATE_MULTI;
```
Comment (hustjieke) [inline storage/tianmu/handler/tianmu_handler.cpp]: 1. Uniform variables name, the same problem to `tianmuDeleteOrUpdate`.

2.Also, add `if` or `check` before the bool variable? Like `check `if_tianmu_delete_or_update`

3. Wrap the expression with brackets, e.g.: 
```
bool_value = (a==b) || (a==c) || ....
```

File: sql/log_event.cc [hunk 3]
```diff
@@ -9681,6 +9681,19 @@ search_key_in_table(TABLE *table, MY_BITMAP *bi_cols, uint key_type)
   KEY *keyinfo;
   uint res= MAX_KEY;
   uint key;
+  /*
+    PK has bugs, not support
+  */
+  bool tianmu_engine = table && table->s && 
+                      (table->s->db_type() ? table->s->db_type()->db_type == DB_TYPE_TIANMU: false);
+  enum_sql_command sqlCommand = SQLCOM_END;
```
Comment (hustjieke) [inline sql/log_event.cc]: Definition the sql command variable in this form:
```
enum_sql_command save_sql_command= lex->sql_command;
```
The SQLCOM_END should not be an init value.

**Comment by hustjieke** [comment]
@konghaiya pls fix the review comment.

**Review by hustjieke — APPROVED** [review]
LGTM

**Pull request merged**